universum/configuration_support.py

```diff
@@ -8,14 +8,133 @@
     "Step",
     "Configuration",
     "Variations",
+    "combine",
     "set_project_root",
     "get_project_root"
 ]
 
 
+# pylint: disable-msg=line-too-long
 class Step:
     """
-    Step is a collection of configurable project-specific data needed for Universum operations.
+    Step is a collection of configurable project-specific data entries needed for Universum functionality. Individual
+    build steps are collected in a :class:`Configuration` object, which supports arithmetic operations for
+    combination and chaining.
+
+    **Step keys**
+
+    Parameters supplied via named attributes during construction are annotated and can be type-checked for safety.
+    Clients may also provide custom parameters in kwargs - these parameters will be stored internally in `dict` and can
+    be retrieved by indexing. The following list enumerates all the keys supported and used by Universum:
```

Comment:
'all the keys supported in Universum configs by default' (for you say earlier that users can define their own keys as well, and it's also kind of supported by Universum)